hbase-examples/src/main/java/org/apache/hadoop/hbase/coprocessor/example/row/stats/recorder/RowStatisticsTableRecorder.java

```diff
@@ -0,0 +1,187 @@
+    } catch (TimeoutException e) {
+      LOG.warn(
+        "Disruptor shutdown timed out after {} ms. Forcing halt. Some row statistics may be lost",
+        DISRUPTOR_SHUTDOWN_TIMEOUT_MS);
+      disruptor.halt();
+      disruptor.shutdown();
+    }
+    bufferedMutator.close();
+  }
+
+  private static BufferedMutator initializeBufferedMutator(Connection conn,
+    Counter rowStatisticsPutFailed) {
+    Configuration conf = conn.getConfiguration();
+    TableRecorderExceptionListener exceptionListener =
+      new TableRecorderExceptionListener(rowStatisticsPutFailed);
+    BufferedMutatorParams params = new BufferedMutatorParams(NAMESPACED_TABLE_NAME)
+      .rpcTimeout(getInt(conf, HConstants.HBASE_RPC_TIMEOUT_KEY, 15_000))
+      .operationTimeout(getInt(conf, HConstants.HBASE_CLIENT_OPERATION_TIMEOUT, 30_000))
+      .setWriteBufferPeriodicFlushTimeoutMs(
+        getLong(conf, ConnectionConfiguration.WRITE_BUFFER_PERIODIC_FLUSH_TIMEOUT_MS, 60_000L))
+      .writeBufferSize(getLong(conf, ConnectionConfiguration.WRITE_BUFFER_SIZE_KEY,
+        ConnectionConfiguration.WRITE_BUFFER_SIZE_DEFAULT))
+      .listener(exceptionListener);
+    BufferedMutator bufferedMutator = null;
+    try {
+      bufferedMutator = conn.getBufferedMutator(params);
+    } catch (IOException e) {
+      LOG.error(
+        "This should NEVER print! ConnectionImplementation#getBufferedMutator(BufferedMutatorParams bmp) does NOT raise IOExceptions",
+        e);
+    }
+    return bufferedMutator;
+  }
+
+  private static Disruptor<RowStatisticsRingBufferEnvelope>
+    initializeDisruptor(BufferedMutator bufferedMutator, Counter rowStatisticsPutFailures) {
+    Disruptor<RowStatisticsRingBufferEnvelope> disruptor =
+      new Disruptor<>(RowStatisticsRingBufferEnvelope::new, DEFAULT_EVENT_COUNT,
+        new ThreadFactoryBuilder().setNameFormat("rowstats.append-pool-%d").setDaemon(true)
+          .setUncaughtExceptionHandler(Threads.LOGGING_EXCEPTION_HANDLER).build(),
+        ProducerType.MULTI, new BlockingWaitStrategy());
+    disruptor.setDefaultExceptionHandler(new RowStatisticsDisruptorExceptionHandler());
+    RowStatisticsEventHandler rowStatisticsEventHandler =
+      new RowStatisticsEventHandler(bufferedMutator, rowStatisticsPutFailures);
+    disruptor.handleEventsWith(new RowStatisticsEventHandler[] { rowStatisticsEventHandler });
+    return disruptor;
+  }
+
+  protected static class TableRecorderExceptionListener
+    implements BufferedMutator.ExceptionListener {
+
+    private final Counter rowStatisticsPutFailures;
+
+    TableRecorderExceptionListener(Counter counter) {
+      this.rowStatisticsPutFailures = counter;
+    }
+
+    public void onException(RetriesExhaustedWithDetailsException exception,
+      BufferedMutator mutator) {
+      long failedPuts = mutator.getWriteBufferSize();
+      for (int i = 0; i < failedPuts; i++) {
+        rowStatisticsPutFailures.increment();
```

Comment:
I think that there's an `Counter#increment(long)` that you can use instead of a for-loop.